app/src/main/java/com/example/weather/CityPickerViewModel.kt

```diff
@@ -1,37 +1,32 @@
 package com.example.weather
 
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.example.weather.network.CityOverview
-import com.example.weather.network.WeatherApi
-import com.example.weather.network.WeatherApiResponse
+import com.example.weather.network.CityRoomDatabase
+import com.example.weather.repository.CityRepository
 import kotlinx.coroutines.launch
 
 const val cityIds = "4772354,5368361,5128581,4887398,4699066,5809844,5856195,5419384,5391959,4684888,5746545,4990729,4335045"
 const val appId = "bd4472d97ca0b479dc32513cf50a1bf3"
 const val units = "metric"
 
-class CityPickerViewModel : ViewModel() {
+class CityPickerViewModel(application: Application) : AndroidViewModel(application){
```

Comment:
this shouldn't take in an application instance, it should be taking in a repository instance

Reply:
Also, you need to add your repository to your application class, similarly to how you added the database to the Application class